tests/scripts/check-generated-files.sh

```diff
@@ -65,6 +65,6 @@ check()
 }
 
 check scripts/generate_errors.pl library/error.c
-check scripts/generate_query_config.pl programs/ssl/query_config.c
+check scripts/generate_query_config.pl programs/test/query_config.c
```

Comment:
Feels more like a `util` program than a `test` program to me, like `strerror`.

Reply:
My thinking was the only program that uses this C file is in test, so it's placed there. Could move to util if you feel strongly about this, but it isn't built as a standalone util (it was previously part of ssl_client2 and ssl_server2 and the test program, but never a standalone util).

Reply:
I don't feel strongly enough to block the PR on it, but your reasoning doesn't make sense to me. `query_config` itself isn't used for testing anything in Mbed Crypto, so why would it be in `test`? If the only reason it's there is that it's being tested, that's not actually a reason to have it and it should be removed. But it's also useful for other purposes, namely to report build information as part of a test campaign or a platform integration. That makes it a `util`.

Reply:
If it built on its own, I'd agree, it'd be a `util`. As-is, it's just a C source file and doesn't have a `main()` function.

Reply:
I still think it's a `util` but that's not important. Ok.